I'm getting this exception when running ./setup.sh --run:
```
[INFO] --- exec-maven-plugin:1.2.1:java (default-cli) @ jboss-eap-application-client ---
        Add a value to App1Cache with the AdminApp and check on the same instance that the value is correct added
[WARNING] 
java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.codehaus.mojo.exec.ExecJavaMojo$1.run(ExecJavaMojo.java:297)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalStateException: EJBCLIENT000025: No EJB receiver available for handling [appName:jboss-eap-application-adminApp, moduleName:ejb, distinctName:] combination for invocation context org.jboss.ejb.client.EJBClientInvocationContext@24148538
	at org.jboss.ejb.client.EJBClientContext.requireEJBReceiver(EJBClientContext.java:798)
	at org.jboss.ejb.client.ReceiverInterceptor.handleInvocation(ReceiverInterceptor.java:128)
	at org.jboss.ejb.client.EJBClientInvocationContext.sendRequest(EJBClientInvocationContext.java:186)
	at org.jboss.ejb.client.EJBInvocationHandler.sendRequestWithPossibleRetries(EJBInvocationHandler.java:255)
	at org.jboss.ejb.client.EJBInvocationHandler.doInvoke(EJBInvocationHandler.java:200)
	at org.jboss.ejb.client.EJBInvocationHandler.doInvoke(EJBInvocationHandler.java:183)
	at org.jboss.ejb.client.EJBInvocationHandler.invoke(EJBInvocationHandler.java:146)
	at com.sun.proxy.$Proxy23.addToApp1Cache(Unknown Source)
	at org.jboss.as.quickstarts.datagrid.eap.app.AdminClient.main(AdminClient.java:104)
	... 6 more
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] JBoss JDG Quickstart: jboss-eap-application - parent SUCCESS [  0.282 s]
[INFO] JBoss JDG Quickstart: jboss-eap-application - AppTwo SUCCESS [  0.012 s]
[INFO] JBoss JDG Quickstart: jboss-eap-application - AppTwo - ejb SUCCESS [  0.304 s]
[INFO] JBoss JDG Quickstart: jboss-eap-application - AppOne SUCCESS [  0.006 s]
[INFO] JBoss JDG Quickstart: jboss-eap-application - AppOne - ejb SUCCESS [  0.038 s]
[INFO] JBoss JDG Quickstart: jboss-eap-application - AppOne - ear SUCCESS [  0.028 s]
[INFO] JBoss JDG Quickstart: jboss-eap-application - AppTwo - ear SUCCESS [  0.018 s]
[INFO] JBoss JDG Quickstart: jboss-eap-application - adminApp SUCCESS [  0.016 s]
[INFO] JBoss JDG Quickstart: jboss-eap-application - adminApp ejb SUCCESS [  0.017 s]
[INFO] JBoss JDG Quickstart: jboss-eap-application - adminApp ear SUCCESS [  0.030 s]
[INFO] JBoss JDG Quickstart: jboss-eap-application - client FAILURE [ 10.913 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------

```

--- a/.gitignore
+++ b/.gitignore
@@ -19,3 +19,7 @@ README*.html
 
 # javascript
 node_modules
+
+#eap-cluster-app
+eap-cluster-app/server.log
+eap-cluster-app/setupOutput

--- a/eap-cluster-app/README.md
+++ b/eap-cluster-app/README.md
@@ -96,14 +96,16 @@ Configure and Start the Servers in standalone mode
 
    Inside security-realms:
 
+```xml
    <security-realm name="ejb-security-realm">
        <server-identities>
            <secret value="cXVpY2stMTIz"/>   <!-- this is a Base64 encoded password for the quickuser user -->
        </server-identities>
    </security-realm>
-
+```
    Inside remoting subsystem:
 
+```xml
    <outbound-connections>
        <remote-outbound-connection name="remote-ejb-connection" outbound-socket-binding-ref="remote-ejb" username="quickuser" security-realm="ejb-security-realm" protocol="http-remoting">
            <properties>
@@ -112,12 +114,15 @@ Configure and Start the Servers in standalone mode
            </properties>
        </remote-outbound-connection>
    </outbound-connections>
+```
 
    Inside socket-binding-group:
 
+```xml
    <outbound-socket-binding name="remote-ejb">
        <remote-destination host="localhost" port="8280"/>
    </outbound-socket-binding>
+```
 
 3. Copy the prepared EAP server to 4 different directories EAP_HOME[1-4].
 

--- a/eap-cluster-app/build.sh
+++ b/eap-cluster-app/build.sh
@@ -32,45 +32,21 @@ case "$1" in
 
       if [ "${1}" == "--setup" ]; #Starting in standalone mode otherwise in domain mode
       then
-         echo "Copying the configurations from README file to EAP ..."
-         # modify EAP standalone.xml according to README
-         # copy <security-realm> from README to standalone.xml
-         sed '/<security-realm name="ejb-security-realm">/,/<\/security-realm>/!d' README.md > tmpbuff
-         sed '/<\/security-realms>/i \
-               INSERT-SECURITY-REALM-HERE' -i "${EAP_HOME}"/standalone/configuration/standalone.xml
-         sed '/INSERT-SECURITY-REALM-HERE/ {
-              r tmpbuff
-              d
-              }' -i "${EAP_HOME}"/standalone/configuration/standalone.xml
+        echo "Copying the configurations from README file to EAP ..."
 
-         sed '/<outbound-connections>/,/<\/outbound-connections>/!d' README.md > tmpbuff
-         sed '/<subsystem xmlns="urn:jboss:domain:remoting:.*">/a \
-            INSERT_OUTBOUND_CONNECTION_HERE' -i "${EAP_HOME}"/standalone/configuration/standalone.xml
-         sed '/INSERT_OUTBOUND_CONNECTION_HERE/ {
-              r tmpbuff
-              d
-              }' -i "${EAP_HOME}"/standalone/configuration/standalone.xml
+        echo "Copying EAP server to directories"
+        cp -R "${EAP_HOME}" setupOutput/eap-server/server1
+        export EAP_HOME1=`cd setupOutput/eap-server/server1;pwd`
 
-         sed '/<outbound-socket-binding.*/,/<\/outbound-socket-binding/!d' README.md > tmpbuff
-         sed '/<\/socket-binding-group>/i \
-               INSERT-SOCKET-BINDING-HERE' -i "${EAP_HOME}"/standalone/configuration/standalone.xml
-         sed '/INSERT-SOCKET-BINDING-HERE/ {
-              r tmpbuff
-              d
-              }' -i "${EAP_HOME}"/standalone/configuration/standalone.xml
+        cp -R "${EAP_HOME}" setupOutput/eap-server/server2
+        export EAP_HOME2=`cd setupOutput/eap-server/server2;pwd`
 
-         echo "Copying EAP server to directories"
-         cp -R "${EAP_HOME}" setupOutput/eap-server/server1
-         export EAP_HOME1=`cd setupOutput/eap-server/server1;pwd`
+        cp -R "${EAP_HOME}" setupOutput/eap-server/server3
+        export EAP_HOME3=`cd setupOutput/eap-server/server3;pwd`
 
-         cp -R "${EAP_HOME}" setupOutput/eap-server/server2
-         export EAP_HOME2=`cd setupOutput/eap-server/server2;pwd`
+        cp -R "${EAP_HOME}" setupOutput/eap-server/server4
+        export EAP_HOME4=`cd setupOutput/eap-server/server4;pwd`
 
-         cp -R "${EAP_HOME}" setupOutput/eap-server/server3
-         export EAP_HOME3=`cd setupOutput/eap-server/server3;pwd`
-
-         cp -R "${EAP_HOME}" setupOutput/eap-server/server4
-         export EAP_HOME4=`cd setupOutput/eap-server/server4;pwd`
 
         echo "Running quckstart in Standalon mode. Starting all 4 instances of the server..."
         sh "${EAP_HOME1}/"bin/standalone.sh -Djboss.node.name=node1 > "${EAP_HOME1}/server.log" &
@@ -85,9 +61,15 @@ case "$1" in
         sh "${EAP_HOME4}"/bin/standalone.sh -Djboss.node.name=node4 -Djboss.socket.binding.port-offset=300 -c standalone-ha.xml > "${EAP_HOME4}/server.log" &
         sleep 10
 
-        echo "Adding the configuration for node2 (AppOne) to use EJB server-to-server invocation ..."
-        "${EAP_HOME2}"/bin/jboss-cli.sh -c --controller=localhost:10090 --file=install-appOne-standalone.cli > setupOutput/install.log
-        sleep 10
+        echo "Adding the configuration for nodes (AppOne) to use EJB server-to-server invocation ..."
+        sh "${EAP_HOME1}"/bin/jboss-cli.sh  -c --controller=localhost:9990 --file=install-appOne-standalone.cli > setupOutput/install_server_1.log &
+        sleep 2
+        sh "${EAP_HOME2}"/bin/jboss-cli.sh  -c --controller=localhost:10090 --file=install-appOne-standalone.cli > setupOutput/install_server_2.log &
+        sleep 2
+        sh "${EAP_HOME3}"/bin/jboss-cli.sh  -c --controller=localhost:10190 --file=install-appOne-standalone.cli > setupOutput/install_server_3.log &
+        sleep 2
+        sh "${EAP_HOME4}"/bin/jboss-cli.sh  -c --controller=localhost:10290 --file=install-appOne-standalone.cli > setupOutput/install_server_4.log &
+        sleep 2
 
         echo "Copying the resources to EAP servers.."
         cp adminApp/ear/target/jboss-eap-application-adminApp.ear "${EAP_HOME1}"/standalone/deployments
@@ -131,15 +113,14 @@ case "$1" in
       (ps -ef | grep 'Process Controller' | grep -v grep | awk '{print $2}' | xargs kill) > /dev/null 2>&1
       (ps -ef | grep 'Standalone' | grep -v grep | awk '{print $2}' | xargs kill) > /dev/null 2>&1
 
-      rm -rf setupOutput/
    ;;
 
    *)
       echo -e "${GREEN}Possible list of arguments is:"
-      echo "--setup sets up the EAP servers, starts them in standalone mode and installs quickstart configuration;"
+      echo "--setup cleanup, sets up the EAP servers, starts them in standalone mode and installs quickstart configuration;"
       echo "--setup-domain sets up the EAP servers, starts them in domain mode and installs quickstart configuration. Use this argument with --setup argument;"
       echo "--run runs the quickstarts;"
-      echo -e "--teardown stops the servers and cleans up the folders;${NO_COLOR}"
+      echo -e "--teardown stops the servers;${NO_COLOR}"
       exit 1;
 esac
 

